dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/DefaultTrackerImportService.java

```diff
@@ -181,6 +188,26 @@ protected void commitBundle( TrackerImportParams params, TrackerImportReport imp
         }
     }
 
+    protected void deleteBundle( TrackerImportParams params, TrackerImportReport importReport, List<TrackerBundle> trackerBundles )
+    {
+        Timer commitTimer = new SystemTimer().start();
+
+        trackerBundles.forEach( tb -> importReport.getBundleReports().add( trackerBundleService.delete( tb ) ) );
+
+        if ( !importReport.isEmpty() )
+        {
+            importReport.setStatus( TrackerStatus.WARNING );
```

Comment:
In case of validation errors neither delete nor commit would be executed. In other cases, execution will flow as it should and if there is any error then it will log it as WARNING without stopping the execution.